### PR TITLE
Update windows build

### PR DESCRIPTION
**Pull request by ksh8281**
* using cmake instead of maintain another files for windows
* delete own ICU build files for windows
* Fix some bugs running on windows

### CODE REVIEW

File: .github/workflows/analysis-actions.yml [hunk 1]
```diff
@@ -64,7 +64,7 @@ jobs:
         sudo apt-get install -y pypy ninja-build libicu-dev gcovr
     - name: Build x64
       env:
-        BUILD_OPTIONS: -DESCARGOT_HOST=linux -DESCARGOT_ARCH=x64 -DESCARGOT_MODE=release -DESCARGOT_THREADING=ON -DESCARGOT_TCO=ON -DESCARGOT_COVERAGE=ON -DESCARGOT_OUTPUT=shell_test -GNinja
+        BUILD_OPTIONS: -DESCARGOT_HOST=linux -DESCARGOT_ARCH=x64 -DESCARGOT_MODE=release -DESCARGOT_THREADING=ON -DESCARGOT_TCO=ON -DESCARGOT_COVERAGE=ON -DESCARGOT_TEMPORAL=ON -DESCARGOT_TCO=ON -DESCARGOT_TEST=ON -DESCARGOT_OUTPUT=shell -GNinja
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply .github/workflows/analysis-actions.yml]: Nice catch!

File: .github/workflows/analysis-actions.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ jobs:
         tar xzf cov-analysis-linux64.tar.gz --strip 1 -C cov-analysis-linux64
     - name: Build
       env:
-        BUILD_OPTIONS: -DESCARGOT_HOST=linux -DESCARGOT_ARCH=x64 -DESCARGOT_MODE=release -DESCARGOT_THREADING=ON -DESCARGOT_TCO=ON -DESCARGOT_CODE_CACHE=ON -DESCARGOT_WASM=ON -DESCARGOT_OUTPUT=shell_test -GNinja
+        BUILD_OPTIONS: -DESCARGOT_HOST=linux -DESCARGOT_ARCH=x64 -DESCARGOT_MODE=release -DESCARGOT_THREADING=ON -DESCARGOT_TCO=ON -DESCARGOT_CODE_CACHE=ON -DESCARGOT_WASM=ON -DESCARGOT_TEMPORAL=ON -DESCARGOT_TCO=ON -DESCARGOT_TEST=ON -DESCARGOT_OUTPUT=shell -GNinja
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply .github/workflows/analysis-actions.yml]: Nice catch!

File: .github/workflows/es-actions.yml [hunk 3]
```diff
@@ -36,25 +36,41 @@ jobs:
...
+        arch: x86
+        sdk: "10.0.20348.0"
+    - uses: lukka/get-cmake@latest
     - name: Build Win32 Release
       run: |
-        cd build/windows/escargot
-        msbuild escargot.sln /p:configuration=release /p:platform=win32
+        (new-object System.Net.WebClient).DownloadFile('https://github.com/abbodi1406/vcredist/releases/download/v0.73.0/VisualCppRedist_AIO_x86_x64.exe','VisualCppRedist_AIO_x86_x64.exe')
+        .\VisualCppRedist_AIO_x86_x64.exe /y
+        CMake -G "Visual Studio 16 2019" -DCMAKE_SYSTEM_NAME=Windows -DCMAKE_SYSTEM_VERSION:STRING="10.0" -DCMAKE_SYSTEM_PROCESSOR=x86 -DESCARGOT_ARCH=x86 -DESCARGOT_MODE=release -Bout/win32_release_ninja/ -DESCARGOT_HOST=windows -DESCARGOT_OUTPUT=shell -DESCARGOT_LIBICU_SUPPORT=ON -DESCARGOT_LIBICU_SUPPORT_WITH_DLOPEN=NO -DESCARGOT_TEST=ON -G Ninja -DCMAKE_C_COMPILER=cl -DCMAKE_CXX_COMPILER=cl -DCMAKE_BUILD_TYPE=release
```
[the remark this answers is not in the document]
Comment (ksh8281) [reply .github/workflows/es-actions.yml]: I see. using off is more natural

**Review by clover2123 — APPROVED** [review]
LGTM

**Pull request merged**